bin/host/src/backend/online.rs

```diff
@@ -86,8 +96,19 @@ where
     /// Set the last hint to be received.
     async fn route_hint(&self, hint: String) -> PreimageOracleResult<()> {
         trace!(target: "host-backend", "Received hint: {hint}");
-        let mut hint_lock = self.last_hint.write().await;
-        hint_lock.replace(hint);
+
+        let parsed_hint =
+            hint.parse::<Hint<C::HintType>>().map_err(|_| PreimageOracleError::KeyNotFound)?;
```

Comment:
Is there a downside to modifying the `HintType` itself to describe whether or not the `HintType` is proactive?

Reply:
Yeah, mostly that the `HintType` would need to hold onto some extra metadata that leaks info about the `HintRouter`. We could do that, but felt a bit nicer to just register proactive hints with the backend, since it's more of an implementation detail of the `HintRouter` trait implementation.